spec/jobs/timeout_stuck_builds_job_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe TimeoutStuckBuildsJob do
+  let(:repository) { FactoryGirl.create(:repository, url: 'git@github.com:square/test-repo.git') }
+  let(:branch) { FactoryGirl.create(:branch, :repository => repository) }
+  let(:build) { FactoryGirl.create(:build, :state => :runnable, :branch_record => branch) }
+
+  subject { TimeoutStuckBuildsJob.perform }
+
+  before do
+    build.build_parts.create!(:kind => :spec, :paths => ["foo", "bar"], :queue => :ci)
+         .build_attempts.create!(:state => :running)
+    build.build_parts.create!(:kind => :cucumber, :paths => ["baz"], :queue => :ci)
+         .build_attempts.create!(:state => :running)
+  end
+
+  describe "#perform" do
+    before do
+      repository.update_attributes(assume_lost_after: 10)
+    end
+
+    context "when a repository has assume_lost_after set" do
+      it "should not stop builds that have yet to reach the limit" do
+        expect { subject }.to change { BuildAttempt.where(state: 'running').count }.by(0)
+      end
+
+      it "should stop builds that have reached the limit" do
+        BuildAttempt.first.update_attributes(started_at: 30.minutes.ago)
+        expect { subject }.to change { BuildAttempt.where(state: 'running').count }.by(-1)
```

Comment:
I dislike global counts like this because it relies on you to know the state of the entire test database and any pollution can conceivably cause the test to fail. Instead I think we should create variables for the two build attempts above and verify that their state changed (or didn't change).